app/src/main/java/com/example/habitsmasher/DaysTracker.java

```diff
@@ -0,0 +1,261 @@
+package com.example.habitsmasher;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Set;
+
+public class DaysTracker {
+    /**
+     * This class is used to represent the days of the week a habit takes place, and contains all functionality in
+     * storing, querying, modifying, and returning what days of the week a habit takes place. This class does not currently handle
+     * storing the days in a database, but is capable of taking in information stored in a database and converting to a DaysTracker object.
+     *
+     * @author: Cameron Matthew
+     * @Version: 1.0
+     * @see Habit
+     */
+
+    //instance variables to represent days of the week
+    private ArrayList<Boolean> _days;
+
+
+    /**
+     * Creates a DaysTracker object with all values set to false. This constructor is the default constructor.
+     */
+    public DaysTracker(){
+        //arraylist of booleans that represent the days of the week. autofilled to 7 bools.
+        _days = new ArrayList<Boolean>(Arrays.asList(new Boolean[7]));
+
+        //fill with all false
+        Collections.fill(_days, Boolean.FALSE);
+    }
+
+
+    /**
+     * Creates a DaysTracker with {@link String} input. The string contains all of the days that are to
+     * be set to true, and all other days false.
+     *<p>
+     * The string takes the format "day1 day2 ... day7", where each day is the first 2 letters of that day, separated by a space. The string can be in any order.
+     * EX) "MO FR WE TH"
+     *
+     * The string is assumed to be formatted correctly, but will not throw any exceptions if not correctly formatted. If any
+     * doubts are present when using this constructor, it may be wise to use the base constructor and set individual days yourself.
+     *
+     * @param days: The string of dates to use, formatted according to above guide.
+     */
+    public DaysTracker(String days){
```

Comment:
This constructor is used in viewing habits, as the String days is the string of days that comes from the DB.